backend/remote-state/http/backend_state.go

```diff
@@ -0,0 +1,177 @@
+package http
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/state"
+	"github.com/hashicorp/terraform/state/remote"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+const (
+	stateFileSuffix = ".tfstate"
+	lockFileSuffix  = ".tflock"
+)
+
+// States get a list of all states.
+func (b *Backend) States() ([]string, error) {
+	var result []string
+	client := &RemoteClient{
+		client:   b.client,
+		address:  b.address,
+		username: b.username,
+		password: b.password,
+	}
+
+	resp, err := client.Get()
+	if err != nil {
+		return nil, err
+	}
+	// Read in the body
+	// Get all the data sent by REST
```

Comment:
can you document this more... and explain the *why*. Why do you need to perform string replacement? and sorting? what are you doing here?